rust/yarp-sys/tests/pack_tests.rs

```diff
@@ -0,0 +1,57 @@
+use std::{ffi::CString, mem::MaybeUninit};
+
+use yarp_sys::{
+    yp_pack_encoding, yp_pack_endian, yp_pack_length_type, yp_pack_parse, yp_pack_result,
+    yp_pack_signed, yp_pack_size, yp_pack_type, yp_pack_variant, yp_size_to_native,
+};
+
+#[test]
+fn pack_parse_test() {
+    let variant_arg = yp_pack_variant::YP_PACK_VARIANT_PACK;
+    let first_format = CString::new("C").unwrap();
+    let end_format = CString::new("").unwrap();
+    let mut format = vec![first_format.as_ptr(), end_format.as_ptr()];
+
+    let mut type_out = MaybeUninit::<yp_pack_type>::uninit();
+    let mut signed_type_out = MaybeUninit::<yp_pack_signed>::uninit();
+    let mut endian_out = MaybeUninit::<yp_pack_endian>::uninit();
+    let mut size_out = MaybeUninit::<yp_pack_size>::uninit();
+    let mut length_type_out = MaybeUninit::<yp_pack_length_type>::uninit();
+    let mut length_out = 0_u64;
+    let mut encoding_out = MaybeUninit::<yp_pack_encoding>::uninit();
+
+    unsafe {
+        let result = yp_pack_parse(
```

Comment:
Amazing. Happy to make this obsolete:

- https://github.com/artichoke/artichoke/pull/2605